src/controllers/dashboard/plugins/repositories.js

```diff
@@ -0,0 +1,153 @@
+import loading from 'loading';
+import libraryMenu from 'libraryMenu';
+import globalize from 'globalize';
+import dialogHelper from 'dialogHelper';
+import 'emby-button';
+import 'emby-checkbox';
+import 'emby-select';
+import 'formDialogStyle';
+import 'listViewStyle';
+
+let repositories = [];
+
+function reloadList(page) {
+    loading.show();
+    ApiClient.getJSON(ApiClient.getUrl('Repositories')).then(list => {
+        repositories = list;
+        populateList({
+            listElement: page.querySelector('#repositories'),
+            noneElement: page.querySelector('#none'),
+            repositories: repositories
+        });
+    }).catch(error => {
+        console.error('error loading repositories');
+        page.querySelector('#none').classList.remove('hide');
+        loading.hide();
+    });
+}
+
+function saveList(page) {
+    loading.show();
+    ApiClient.ajax({
+        type: 'POST',
+        url: ApiClient.getUrl('Repositories'),
+        data: JSON.stringify(repositories),
+        contentType: 'application/json'
+    }).then(response => {
+        reloadList(page);
+    }).catch(error => {
+        console.error('error saving repositories');
+        loading.hide();
+    });
+}
+
+function populateList(options) {
+    var html = '';
+
+    html += '<div class="paperList">';
+    for (var i = 0; i < options.repositories.length; i++) {
+        html += getRepositoryHtml(options.repositories[i]);
+    }
+
+    html += '</div>';
+    if (!options.repositories.length) {
+        options.noneElement.classList.remove('hide');
+    }
+
+    options.listElement.innerHTML = html;
+    loading.hide();
+}
+
+function getRepositoryHtml(repository) {
+    var html = '';
+
+    html += '<div class="listItem listItem-border">';
+    html += `<a is="emby-linkbutton" style="margin:0;padding:0" class="clearLink listItemIconContainer" href="${repository.Url}">`;
+    html += '<span class="material-icons listItemIcon open_in_new"></span>';
+    html += '</a>';
+    html += '<div class="listItemBody two-line">';
+    html += `<h3 class='listItemBodyText'>${repository.Name}</h3>`;
+    html += `<div class="listItemBodyText secondary">${repository.Url}</div>`;
+    html += '</div>';
+    html += `<button type="button" is="paper-icon-button-light" id="${repository.Url}" class="btnDelete" title="${globalize.translate('ButtonDelete')}"><span class="material-icons delete"></span></button>`;
+    html += '</div>';
```

Comment:
Although I know it might be tempting to not scape inner quotes, I think we shouldn't do this for the sake of consistency. We merged #1069 to be consistent and we're breaking that if we merge this in this state imho:

```suggestion
    html += '<div class="listItem listItem-border">';
    html += '<a is="emby-linkbutton" style="margin:0;padding:0" class="clearLink listItemIconContainer" href="${repository.Url}">';
    html += '<span class="material-icons listItemIcon open_in_new"></span>';
    html += '</a>';
    html += '<div class="listItemBody two-line">';
    html += '<h3 class="listItemBodyText">${repository.Name}</h3>';
    html += '<div class="listItemBodyText secondary">${repository.Url}</div>';
    html += '</div>';
    html += '<button type="button" is="paper-icon-button-light" id="${repository.Url}" class="btnDelete" title="${globalize.translate(\'ButtonDelete\')}"><span class="material-icons delete"></span></button>';
    html += '</div>';
```

We have some of those backticks left in our code but, afaik, we haven't reached a conclusion on our style rules for this. In case we agree it doesn't mind, I'll greencheck this, but I just want to be sure first :)

Reply:
I actually agree with you on this one, but other people have started using the backticks elsewhere in new code, so I've been using them too.

Reply:
I thought we did reach the conclusion of "prefer template literals".

Imo, escaping should be avoided when possible. The single quote rule already takes this into account (allowing double quotes when single quotes would need to be escaped).

Here, replacing the backticks (template literals) by single quotes breaks the template literal, so ${variable} will show up as text and won't resolve to it's proper value.

Imo, the proper fix is this:
```suggestion
    html += '<div class="listItem listItem-border">';
    html += `<a is="emby-linkbutton" style="margin:0;padding:0" class="clearLink listItemIconContainer" href="${repository.Url}">`;
    html += '<span class="material-icons listItemIcon open_in_new"></span>';
    html += '</a>';
    html += '<div class="listItemBody two-line">';
    html += `<h3 class="listItemBodyText">${repository.Name}</h3>`;
    html += `<div class="listItemBodyText secondary">${repository.Url}</div>`;
    html += '</div>';
    html += `<button type="button" is="paper-icon-button-light" id="${repository.Url}" class="btnDelete" title="${globalize.translate('ButtonDelete')}"><span class="material-icons delete"></span></button>`;
    html += '</div>';
```

(ie fix the single quotes used in the HTML properties on one of the lines, probably by error.

Reply:
Great then, good to know! 👍🏻

@dkanada commit @MrTimscampi suggestion and this is good to go imo.